frontend/src/components/auth/Login.tsx

```diff
@@ -48,16 +67,23 @@ const Login = (): React.ReactElement => {
       password,
       login,
     );
+    if (user === null) {
+      setLoginFail(true);
+    }
     setAuthenticatedUser(user);
   };
 
   const onLogInClick = async () => {
     switch (loginState) {
       case LoginState.EnterEmail:
-        setLoginState(LoginState.EnterPassword);
+        if (emailRef.current?.validity.valid) {
```

Comment:
If this is not true, this leaves no indication to the user why their validation failed. Could you add an extra error state where the email is incorrect, telling them that what they entered isn't a valid email?